src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -412,6 +412,16 @@ internal static class ConstraintsHelper
                     containingSymbol.CheckConstraintTypeVisibility(syntax.Location, constraintType, diagnostics);
                     constraintTypeBuilder.Add(constraintType);
                 }
+
+                if (constraintType.ContainsNullableReferenceTypes())
+                {
+                    bool onLocalFunction = containingSymbol.Kind == SymbolKind.Method && ((MethodSymbol)containingSymbol).MethodKind == MethodKind.LocalFunction;
+                    containingSymbol.DeclaringCompilation.EnsureNullableAttributeExists(diagnostics, syntax.Location, modifyCompilation: !onLocalFunction);
```

Comment:
>modifyCompilation [](start = 118, length = 17)

@jaredpar Good catch. Thanks
I investigated this further and this PR indeed has a bug. 
For the ReadOnly attribute, we have `LocalRewriter.VisitLocalFunctionStatement` register uses of `in`.
I need to do the same for the Nullable attribute.
The tests in this PR are correct, but we still have logic that synthesizes the Nullable attribute whenever LangVersion is 8, which hides this problem.
I'll fix.

Reply:
Fixed now.
I've verified the fix in my other branch (where Nullable is synthesized only when required by usage of `?` annotation).